tools/external_ic.F90

```diff
@@ -937,8 +955,15 @@ subroutine read_gfs_ic()
           if (data_source_fv3gfs) call register_restart_field(GFS_restart, 't', temp, dim_names_3d3, is_optional=.true.)
 
           ! prognostic tracers
+
           do nt = 1, ntracers
-             q(:,:,:,nt) = -999.99
+
+             IF ( .not. ( nt == Atm%flagstruct%nwat .and. Atm%flagstruct%nwat == 7 ) ) THEN
```

Comment:
This is a necessary trigger to ensure proper initialization of tracer quantities when the experiment defines more tracers than are available in the IC files.  The field_table has attribute modifiers that allow one to specify the initial profile for a tracer.  By having the temporary 'q' define all non-existent IC tracers to zero, you are replacing any initial field_table profiles defined at lines 639-648 for Atm%q and Atm%qdiag.  

If you have non-IC file tracers you want to be zero, the proper method is to use the attributes correctly within the field_table.  The -999.99 value works with the logic at line 3087 to the non-existent IC tracers and also ensures there is no potential round-off errors brought in by the remapping and 'fill' process.  If you still have questions, we can discuss further.

Reply:
The negative value initialization seems to have been added some time in the past year or so? It is only the hail mass that has a problem here, since that is not in the inputs I have used. Is there a reason not to simply set everything to zero, or was this meant to trap some particular issue?

Reply:
Ah, OK, thanks for the explanation. I will revert that and test with changing the field_table.

Reply:
As stated above, not all tracer initial values should be set to zero.  The q(:)=0. logic you added forces all tracers to be zero if they don't have an entry in the IC files.  If you want the hailwat tracer to be defined as zero, a properly configured entry for the hail tracer in the field_table will ensure it's initial value in Atm%q will be so and the value of q(halwat) = -999.99 and the if-test at line 3087 is left in place.